src/app/qgsmaptoolreshape.cpp

```diff
@@ -182,29 +185,13 @@ void QgsMapToolReshape::reshape( QgsVectorLayer *vlayer )
           QHash<QgsVectorLayer *, QSet<QgsFeatureId> > ignoreFeatures;
           ignoreFeatures.insert( vlayer, vlayer->allFeatureIds() );
 
-          QList<QgsVectorLayer *>  avoidIntersectionsLayers;
-          switch ( QgsProject::instance()->avoidIntersectionsMode() )
+          const QgsAvoidIntersectionsOperation::Result res = avoidIntersections.apply( vlayer, f.id(), geom, ignoreFeatures );
+          if ( res.operationResult == Qgis::GeometryOperationResult::InvalidInputGeometryType )
           {
-            case Qgis::AvoidIntersectionsMode::AvoidIntersectionsCurrentLayer:
-              avoidIntersectionsLayers.append( vlayer );
-              break;
-            case Qgis::AvoidIntersectionsMode::AvoidIntersectionsLayers:
-              avoidIntersectionsLayers = QgsProject::instance()->avoidIntersectionsLayers();
-              break;
-            case Qgis::AvoidIntersectionsMode::AllowIntersections:
-              break;
-          }
-          Qgis::GeometryOperationResult res = Qgis::GeometryOperationResult::NothingHappened;
-          if ( avoidIntersectionsLayers.size() > 0 )
-          {
-            res = geom.avoidIntersectionsV2( QgsProject::instance()->avoidIntersectionsLayers(), ignoreFeatures );
-            if ( res == Qgis::GeometryOperationResult::InvalidInputGeometryType )
-            {
-              emit messageEmitted( tr( "An error was reported during intersection removal" ), Qgis::MessageLevel::Critical );
-              vlayer->destroyEditCommand();
-              stopCapturing();
-              return;
-            }
+            emit messageEmitted( tr( "An error was reported during intersection removal" ), Qgis::MessageLevel::Critical );
```

Comment:
Can we make this a warning? I think yellow is severe enough for this.

Reply:
Yes. Done